internal/jimm/bootstrap/bootstrap.go

```diff
@@ -198,23 +211,41 @@ func (b *bootstrapManager) StartBootstrap(ctx context.Context, user *openfga.Use
 	return jobId.String(), nil
 }
 
-// BootstrapExecutor holds a wrapper to run a command. It is primarily for testing
-// and when running the bootstrap command the implemention to be used is [DefaultBootstrapExecutor].
-type BootstrapExecutor interface {
-	RunWrapper(
-		ctx context.Context,
-		binaryPath, jujuDataDir string,
-		params jujucommands.BootstrapCmdParams,
-	) (<-chan jujucommands.OutputLine, jujuclient.ClientStore, func(), error)
+type command struct {
+	binaryPath  string
+	jujuDataDir string
+}
+
+// Bootstrap implements the JujuCommands interface.
+func (c command) Bootstrap(ctx context.Context, p jujucommands.BootstrapCmdParams) (
+	<-chan jujucommands.OutputLine, jujuclient.ClientStore, func(), error) {
+
+	r := jujucommands.NewCommandRunner(c.binaryPath, c.jujuDataDir)
+	return jujucommands.NewBootstrapCmd(r).Run(ctx, p)
+}
+
+// DestroyController implements the JujuCommands interface.
+func (c command) DestroyController(ctx context.Context, p jujucommands.DestroyControllerCmdParams) (
+	<-chan jujucommands.OutputLine, error) {
+
+	r := jujucommands.NewCommandRunner(c.binaryPath, c.jujuDataDir)
+	return jujucommands.NewDestroyControllerCmd(r).Run(ctx, p)
 }
 
-// DefaultBootstrapExecutor defines the default, and expected bootstrap executor
+// commandFactory provides a concrete implementation of [CommandFactory]
 // to be used in a [bootstrapManager.BootstrapJob]
-type DefaultBootstrapExecutor struct{}
+type commandFactory struct{}
```

Comment:
For @ale8k, I realise this struct has the same name as the interface it implements. I think that's fine and would prefer not to change it.

Reply:
Yeah that's ok.